src/test/java/uk/gov/companieshouse/api/testdata/controller/TestUsersControllerTest.java

```diff
@@ -0,0 +1,109 @@
+package uk.gov.companieshouse.api.testdata.controller;
+
+import org.junit.jupiter.api.Test;
+import org.junit.jupiter.api.extension.ExtendWith;
+import org.mockito.InjectMocks;
+import org.mockito.Mock;
+import org.mockito.junit.jupiter.MockitoExtension;
+import org.springframework.http.HttpStatus;
+import org.springframework.http.ResponseEntity;
+import uk.gov.companieshouse.api.testdata.exception.DataException;
+import uk.gov.companieshouse.api.testdata.model.rest.UserTestData;
+import uk.gov.companieshouse.api.testdata.model.rest.UsersSpec;
+import uk.gov.companieshouse.api.testdata.service.UsersTestDataService;
+
+import java.util.Map;
+import java.util.Objects;
+
+import static org.junit.jupiter.api.Assertions.*;
+import static org.junit.jupiter.api.Assertions.assertThrows;
+import static org.mockito.ArgumentMatchers.any;
+import static org.mockito.Mockito.*;
+
```

Comment:
Do we have any tests to prove that the user / roles service is not accessible to the public?

Reply:
we are injecting existing InternalUserInterceptor and it has unit tests in that library.